README.md

```diff
@@ -102,6 +102,30 @@ npm install instantsearch.js algoliasearch
 yarn add instantsearch.js algoliasearch
 ```
 
+### TypeScript users
+
+To use InstantSearch in a TypeScript environment, depending on your `algoliasearch` version, different types must be imported:
+
+In case you are not using `algoliasearch` as the search client, it must still be installed in TypeScript environments.
+
+#### `algoliasearch` version 4
+
+V4 uses the types provided by both `algoliasearch` and `@algolia/client-search`.
+
+```bash
+yarn add algoliasearch@4 @algolia/client-search
+```
+
+#### `algoliasearch` version 3
+
+```bash
+npm install @types/algoliasearch@3
+# or
```

Comment:
Why do we document npm here and not for the other snippets?

Reply:
```suggestion
```